### PR TITLE
Add watch app target to the sample project.

### PR DESCRIPTION
**Pull request by qyang-nj**
This PR adds watch app target to the sample project. It works with both Buck CLI build and Buck generated project.

Based on observation, each BUCK file will be generated into a `xcodeproj`. However, for the watch app, it has to be in the main `xcodeproj` as a separate target. The key is to make sure the watch rules and the main app rules are in the same BUCK file.

image

### CODE REVIEW

File: App/BUCK [hunk 1]
```diff
@@ -99,9 +99,79 @@ apple_bundle(
     product_name = "ExampleApp",
     info_plist = "Info.plist",
     info_plist_substitutions = info_plist_substitutions("ExampleApp"),
+    deps = [
+        ":DemoWatchApp#watch",
```
Comment (dfed) [inline App/BUCK]: A comment would be great about why `#watch` is needed here. Maybe with a link to existing Buck documentation (assuming that some exists).

File: App/Watch/WatchExtension/ExtensionDelegate.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
```
[the remark this answers is not in the document]
Comment (qyang-nj) [reply App/Watch/WatchExtension/ExtensionDelegate.h]: My PR is based on Michael's PR. Those objc code was brought from [Facebook's test code](https://github.com/facebook/buck/tree/a6c7d7391a7a8e575c6372cac6690a240b5907ea/test/com/facebook/buck/apple/testdata/watch_application_bundle). I think it's worth another PR to replace those with Swift code in separate PR.

[the remark this answers is not in the document]
Comment (dfed) [reply App/Watch/WatchExtension/ExtensionDelegate.h]: Ah! I'd missed that PR 😬. Happy to merge the watch code as-is then. Thank you for the explanation 🙂

File: Config/configs.bzl [hunk 3]
```diff
@@ -83,3 +83,16 @@ def info_plist_substitutions(name):
         "PRODUCT_NAME": name,
     }
     return substitutions
+
+def watch_binary_configs():
```
Comment (dfed) [inline Config/configs.bzl]: nice!

**Comment by bachand** [comment]
🎉

**Comment by fdiaz** [comment]
👏

**Pull request merged**